CMakeLists.txt

```diff
@@ -1,7 +1,19 @@
 cmake_minimum_required(VERSION 3.8)
 project(tarxx VERSION 0.1.0 LANGUAGES CXX)
 
+IF (WIN32)
+    message(FATAL_ERROR Windows is not support at the moment)
+ENDIF()
+
+option(WITH_LZ4 "Set to ON to enable LZ4 compression support" OFF)
+option(WITH_COMPRESSION "Set to ON to enable compression support without any algorithms" OFF)
```

Comment:
hmmm... now that gets tricky... `WITH_LZ4=ON` should require `WITH_COMPRESSION=ON` (or `WITH_LZ4` being only available if `WITH_COMPRESSION=ON`)  ... or at least we need to bail if that invariant is broken.

can you please either model the option dependency or make sure we fail configuration if it's inconsistent?

Reply:
failing with compression is off